spark-indexer/src/main/java/org/uniprot/store/spark/indexer/proteome/ProteomeDocumentsToHPSWriter.java

```diff
@@ -0,0 +1,106 @@
+package org.uniprot.store.spark.indexer.proteome;
+
+import static org.uniprot.store.spark.indexer.common.util.SparkUtils.getCollectionOutputReleaseDirPath;
+
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.Optional;
+import org.apache.spark.api.java.function.Function;
+import org.uniprot.core.proteome.ProteomeEntry;
+import org.uniprot.core.proteome.ProteomeStatistics;
+import org.uniprot.core.taxonomy.TaxonomyEntry;
+import org.uniprot.store.search.SolrCollection;
+import org.uniprot.store.search.document.proteome.ProteomeDocument;
+import org.uniprot.store.spark.indexer.common.JobParameter;
+import org.uniprot.store.spark.indexer.common.util.SolrUtils;
+import org.uniprot.store.spark.indexer.common.writer.DocumentsToHPSWriter;
+import org.uniprot.store.spark.indexer.proteome.mapper.ProteomeEntryToProteomeDocumentMapper;
+import org.uniprot.store.spark.indexer.proteome.mapper.StatisticsToProteomeDocumentMapper;
+import org.uniprot.store.spark.indexer.proteome.mapper.TaxonomyToProteomeDocumentMapper;
+import org.uniprot.store.spark.indexer.proteome.reader.ProteomeStatisticsReader;
+import org.uniprot.store.spark.indexer.taxonomy.reader.TaxonomyRDDReader;
+
+import scala.Tuple2;
+
+public class ProteomeDocumentsToHPSWriter implements DocumentsToHPSWriter {
+    private final JobParameter jobParameter;
+    private final ProteomeStatisticsReader proteomeStatisticsReader;
+    private final TaxonomyRDDReader taxonomyRDDReader;
+    private final ProteomeRDDReader proteomeRDDReader;
+
+    public ProteomeDocumentsToHPSWriter(JobParameter jobParameter) {
+        this.jobParameter = jobParameter;
+        this.proteomeRDDReader = new ProteomeRDDReader(jobParameter, true);
+        this.proteomeStatisticsReader = new ProteomeStatisticsReader(jobParameter);
+        this.taxonomyRDDReader = new TaxonomyRDDReader(jobParameter, true);
+    }
+
+    @Override
+    public void writeIndexDocumentsToHPS() {
+        JavaPairRDD<String, ProteomeDocument> idProteomeDocumentJavaPairRDD =
+                loadProteomeRDD().mapValues(getEntryToProteomeDocumentMapper());
+
+        idProteomeDocumentJavaPairRDD = joinStatistics(idProteomeDocumentJavaPairRDD);
+        idProteomeDocumentJavaPairRDD = joinTaxonomy(idProteomeDocumentJavaPairRDD);
+
+        saveToHPS(idProteomeDocumentJavaPairRDD.values());
+    }
+
+    private JavaPairRDD<String, ProteomeDocument> joinTaxonomy(
+            JavaPairRDD<String, ProteomeDocument> proteomeIdProteomeDocumentJavaPairRDD) {
+        JavaPairRDD<String, String> taxIdProteomeIdJavaRDD =
```

Comment:
// variable name tells it but please put comment also about values in pair to make it more clear
        // <organismId, proteomeId>